packages/compiler/src/parser/state/config.ts

```diff
@@ -0,0 +1,33 @@
+import yaml from 'yaml'
+
+import { Parser } from '..'
+import PARSER_ERRORS from '../../error/errors'
+
+export function config(parser: Parser) {
+  const start = parser.index
+  parser.eat('---')
+
+  // Read until there is a line break followed by a triple dash
+  const data = parser.readUntil(/\n\s*\-\-\-\s*/)
+  console.log('data', data)
+
+  parser.allowWhitespace()
+  parser.eat('---', true)
```

Comment:
again?

Reply:
Yes, it must start and finish with `---`
```md
---
foo: bar
---
Actual markdown content
```